flow/raster_cache_key.h

```diff
@@ -6,26 +6,64 @@
 #define FLUTTER_FLOW_RASTER_CACHE_KEY_H_
 
 #include <unordered_map>
+#include <utility>
+#include <vector>
 
 #include "flutter/fml/hash_combine.h"
 #include "flutter/fml/logging.h"
 #include "third_party/skia/include/core/SkMatrix.h"
 
 namespace flutter {
 
-enum class RasterCacheKeyType { kLayer, kPicture, kDisplayList };
+class RasterCacheKeyID {
+ public:
+  RasterCacheKeyID(const std::vector<uint64_t> ids) : ids_(ids) {}
+
+  const std::vector<uint64_t>& ids() { return ids_; }
+
+  std::size_t GetHash() const {
+    std::size_t seed = fml::HashCombine();
+    for (auto id : ids_) {
+      fml::HashCombineSeed(seed, id);
+    }
+    return seed;
+  }
+
+  bool operator==(const RasterCacheKeyID& other) const {
+    return ids_ == other.ids_;
+  }
+
+  bool operator!=(const RasterCacheKeyID& other) const {
+    return !operator==(other);
+  }
+
+ private:
+  const std::vector<uint64_t> ids_;
+};
+
+enum class RasterCacheKeyType {
+  kLayer,
+  kPicture,
+  kDisplayList,
+  kLayerChildren
+};
 
 enum class RasterCacheKeyKind { kLayerMetrics, kPictureMetrics };
 
 class RasterCacheKey {
  public:
   RasterCacheKey(uint64_t id, RasterCacheKeyType type, const SkMatrix& ctm)
-      : id_(id), type_(type), matrix_(ctm) {
+      : RasterCacheKey(RasterCacheKeyID({id}), type, ctm) {}
```

Comment:
How much does this automatic vectorization of the IDs cost us (time-wise) for the cases where we are just caching the kLayer and not the children?

Reply:
I tested it with the following code:
```
  auto start = fml::TimePoint::Now();
  for (int i = 0; i < 100000; i++) {
    RasterCacheKey raster_key(5, RasterCacheKeyType::kLayer, matrix);
  }
  auto end = fml::TimePoint::Now();
  FML_DLOG(INFO) << "cost " << (end - start).ToNanoseconds();
```
with automatic vectorization：total 117960062 ns， Average generation time per key is 1179 ns.   
Assign directly to an `int64_t` as before：28885147 ns, Average generation time per key is 288 ns.